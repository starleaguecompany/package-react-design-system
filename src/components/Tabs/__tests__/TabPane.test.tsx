import * as React from 'react';
import { render, screen } from '@testing-library/react';

import { Tabs } from '../../../index';

describe('TabPane', () => {
  test('should render correctly with no props', () => {
    render(<Tabs.TabPane />);

    expect(screen.getByTestId('TabPane')).toMatchSnapshot();
    expect(screen.getByTestId('TabPane')).toHaveClass('pane');
  });

  test('should render correctly with attributes', () => {
    render(<Tabs.TabPane id="test-id" className="test-class" />);

    expect(screen.getByTestId('TabPane')).toMatchSnapshot();
    expect(screen.getByTestId('TabPane')).toHaveAttribute('id', 'test-id');
    expect(screen.getByTestId('TabPane')).toHaveClass('test-class');
  });

  test('should render correctly with title prop', () => {
    render(<Tabs.TabPane title="Title pane" />);

    expect(screen.getByTestId('TabPane')).toMatchSnapshot();
    expect(screen.getByTestId('TabPane')).toHaveAttribute('title', 'Title pane');
    expect(screen.getByTestId('TabPane')).toBeVisible();
  });

  test('should render correctly with content', () => {
    render(<Tabs.TabPane title="Title pane">Pane content</Tabs.TabPane>);

    expect(screen.getByTestId('TabPane')).toMatchSnapshot();
    expect(screen.getByTestId('TabPane')).toHaveTextContent('Pane content');
    expect(screen.getByText('Pane content')).toBeVisible();
  });
});
