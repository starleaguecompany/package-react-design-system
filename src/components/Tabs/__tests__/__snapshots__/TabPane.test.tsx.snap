// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPane should render correctly with attributes 1`] = `
<div
  class="test-class pane"
  data-qa="TabPane"
  id="test-id"
/>
`;

exports[`TabPane should render correctly with content 1`] = `
<div
  class="pane"
  data-qa="TabPane"
  title="Title pane"
>
  Pane content
</div>
`;

exports[`TabPane should render correctly with no props 1`] = `
<div
  class="pane"
  data-qa="TabPane"
/>
`;

exports[`TabPane should render correctly with title prop 1`] = `
<div
  class="pane"
  data-qa="TabPane"
  title="Title pane"
/>
`;
